src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { CenterComponent, FormErrorText, StyledForm } from "@/components/Form/formStyling";
+import Button from "@mui/material/Button";
+import {
+    checkErrorOnSubmit,
+    Errors,
+    FormErrors,
+    setError,
+    setField,
+} from "@/components/Form/formFunctions";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBuildingCircleArrowRight } from "@fortawesome/free-solid-svg-icons";
+import RefreshPageButton from "@/app/admin/RefreshPageButton";
+import { Schema } from "@/database_utils";
+import NameCard from "@/app/admin/createCollectionCentre/createCollectionCentreFormSections/NameCard";
+import AcronymCard from "@/app/admin/createCollectionCentre/createCollectionCentreFormSections/AcronymCard";
+import supabase from "@/supabaseClient";
+
+const initialFields: Schema["collection_centres"] = {
+    name: "",
+    acronym: "",
+};
+
+const initialFormErrors: FormErrors = {
+    name: Errors.initial,
+    acronym: Errors.initial,
+};
+
+const formSections = [NameCard, AcronymCard];
+
+const CreateCollectionCentreForm: React.FC<{}> = () => {
+    const [fields, setFields] = useState(initialFields);
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    const [refreshRequired, setRefreshRequired] = useState(false);
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitDisabled(true);
+
+        if (checkErrorOnSubmit(formErrors, setFormErrors)) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        const response = await supabase.from("collection_centres").insert(fields);
+
+        if (response.error) {
+            setSubmitError(Errors.external);
+            setSubmitErrorMessage(response.error.message);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        setSubmitError(Errors.none);
+        setSubmitErrorMessage("");
+        setSubmitDisabled(false);
+        setRefreshRequired(true);
+    };
+
+    return (
+        <CenterComponent>
+            <StyledForm>
+                {formSections.map((Card, index) => {
+                    return (
```

Comment:
haven't tried it out for this but I think you don't need the return if you don't have the curly brackets as well? 

`{formSections.map((Card, index) => <Card key={index} ... `